### PR TITLE
Bug/form validation

### PR DESCRIPTION
**Pull request by ab7289**
This PR fixes issues with form validations such as errors not being displayed or breaking values passing validation and causing errors when saved to the database.

The specific issues from testing day are linked in the attached story, but feel free to test further to make sure something wasn't missed.

### CODE REVIEW

**Comment by mok232** [comment]
This [test](https://github.com/gcivil-nyu-org/fall2019-cs-gy-6063-team-five_fries/issues/268) seems to be missing.

**Comment by mok232** [comment]
This [issue](https://github.com/gcivil-nyu-org/fall2019-cs-gy-6063-team-five_fries/issues/276) mentioned that the suite numbers were allowed to be negative. Since we are allowing the suite numbers to be strings, I don't know if there's an obvious way of dealing with this.

File: mainapp/forms.py [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
 class SiteUserSignupForm(SignupForm):
     email = forms.EmailField()
-    full_name = forms.CharField(max_length=255, label="Full Name")
+    full_name = forms.CharField(
+        max_length=255,
+        label="Full Name",
+        validators=[
+            RegexValidator(
+                regex="^[a-zA-Z\.\ \-]*$",  # noqa W605
+                message="Please only use letters, spaces ( ), periods (.), or dashes (-)",
```
Comment (mok232) [inline mainapp/forms.py]: 👍

File: mainapp/forms.py [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
     work_location = forms.CharField(max_length=255, label="Work Location")
     phone_number = PhoneNumberField(required=False)
+    username = forms.CharField(
+        max_length=20,
+        required=True,
+        validators=[
+            RegexValidator(
+                regex="^[a-zA-Z0-9\.]*$",  # noqa W605
+                message="Please only use alphanumeric characters or periods",
+                code="invalid_username",
```
Comment (mok232) [inline mainapp/forms.py]: 👍

**Comment by ab7289** [comment]
> This [test](https://github.com/gcivil-nyu-org/fall2019-cs-gy-6063-team-five_fries/issues/268) seems to be missing.

Added!

**Comment by ab7289** [comment]
> Handles most of the error validations pretty well especially the Regex validators. Should we also add some error checking for the image upload fields?

I tested it and the image field errors are displaying when you try to upload the wrong file type. I'm not sure why they weren't before.

**Comment by ab7289** [comment]
> This [issue](https://github.com/gcivil-nyu-org/fall2019-cs-gy-6063-team-five_fries/issues/276) mentioned that the suite numbers were allowed to be negative. Since we are allowing the suite numbers to be strings, I don't know if there's an obvious way of dealing with this.

Added!

**Comment by alldne** [comment]
> Not sure if this is because of this PR but the filter sizes seem to have shrunk

I also took a look and I guess it might be a bug of crispy form. Apparently, adding max_value breaks styling.
```
    # No problem
    min_price = forms.IntegerField(min_value=1, required=False)

    # Problem
    min_price = forms.IntegerField(min_value=1, max_value=100000, required=False)
```

**Comment by ab7289** [comment]
> Not sure if this is because of this PR but the filter sizes seem to have shrunk
> 
> image

Fixed!

**Comment by ab7289** [comment]
> Overall LGTM but it seems the styling issue needs to be fixed

Fixed!

**Comment by ab7289** [comment]
Screen Shot 2019-12-03 at 12 40 27 PM
Screen Shot 2019-12-03 at 12 41 02 PM

**Pull request merged**